### PR TITLE
(refactor) some opportunities to use Multi-Catch

### PR DESCRIPTION
**Pull request by desmorto**
- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

**Comment by s1monw** [comment]
@elastimachine test this please

File: core/src/main/java/org/elasticsearch/common/inject/internal/ConstructionContext.java [hunk 1]
```diff
@@ -109,12 +109,10 @@ public Object invoke(Object proxy, Method method, Object[] args)
...
                 // This appears to be not test-covered
                 return method.invoke(delegate, args);
-            } catch (IllegalAccessException e) {
-                throw new RuntimeException(e);
-            } catch (IllegalArgumentException e) {
-                throw new RuntimeException(e);
-            } catch (InvocationTargetException e) {
+            } catch(InvocationTargetException e) {
                 throw e.getTargetException();
+            } catch(IllegalAccessException | IllegalArgumentException e) {
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/common/inject/internal/ConstructionContext.java]: Can you add a space between `catch` and `(` throughout?

**Review by jasontedor — CHANGES_REQUESTED** [review]
I left a comment that applies everywhere.

File: core/src/main/java/org/elasticsearch/common/inject/internal/ConstructionContext.java (hunk 1 above)
Comment (desmorto) [reply core/src/main/java/org/elasticsearch/common/inject/internal/ConstructionContext.java]: of course.

**Comment by desmorto** [comment]
Sorry for that. I updated my engine to add that missing spaces. Thanks for you feedback!

**Comment by jasontedor** [comment]
@desmorto Patches like this and the others that you've submitted where I think you are going through with an IDE and refactoring for the sake of refactoring are generally not helpful. They cost maintainers time without adding much. We have to worry about making sure the code compiles, that the code meets our style, that tests pass, whether or not and where to backport (to save ourselves from merge conflicts on other backports later), run time-consuming tests before backporting, etc., yet they're not really solving a problem. We have a long list of open issues, many that are marked with "adoptme" and "low hanging fruit". These are great places for contributors to start but I would prefer that we not take the time to integrate changes like the ones that you're submitting. I appreciate you taking the time to work with our codebase, and I'm sure it's going to be discouraging to receive this message, but I want to give you the straight feedback before you continue to submit more PRs that we are very unlikely to accept.

We genuinely would be happy to collaborate with you on any of the adoptme issues that you might be interested in.

Relates #26925, #26934, #26935

**Pull request closed without merging**